resourcex/README.md

```diff
@@ -0,0 +1,243 @@
+# resourcex
+
+The `resourcex` package extends the `github.com/pulumi/pulumi/sdk/v3/go/common/resource` package with helpers
+for working with property values.
+
+1. `Extract` - Extract structured values from a property map, with tracking of unknownness and secretness.
+2. `Decode` - Decode a property map into a JSON-like structure containing only values.
```

Comment:
I don't understand why we need both `Decode` and `Extract`. Shouldn't `Decode` be equivalent to:
```go
var dst map[string]any
_, err := Extract(props, &dst)
contract.AssertNoError(err)
return dst
```

Reply:
`Extract` is implemented using `Decode` to obtain a map, then using the `mapstructure` package to map to a Go struct while summarizing the unknownness and secretness.

Reply:
Let me rephrase, I don't understand why we need to make `Extract` and `Decode` public, since `Extract` provides a superset of `Decode`.